### DISCUSSION TITLE
Restrict a tenant token to see a subset of the `displayedAttributes`

### DISCUSSION THREAD

**Discussion by gmourier** [meilisearch/product#636] [Feedback & Feature Proposal]
# Use case

In the situation where I create tenant tokens using Meilisearch in a multi-tenancy use case, I want certain types of users not to be able to see specific fields within the retrieved documents.

# Workarounds

- Do a custom curation depending on the user's type before displaying the search results.
- Maintain an index per tenant token type with different displayed attributes for each index.

# Ideas

Goal: I want to hide the `partner_price` field for non-partner end-users interacting with the search.

When creating a tenant token, I could override the `displayedAttributes` index setting; the drawback is that I must list every displayable attributes while omitting the one I want to hide.

```jsonc
{
      "product": {
             "displayedAttributes": ["name", "price"] //partner_price is hidden since not listed,
             ...
      }
}
```

Another solution could be having a proper tenant token property to hide some fields from the `displayedAttributes` index settings

```jsonc
{
      "product": {
             "hiddenAttributes": ["partner_price"],
             ...
      }
}
```

**Comment by derekperkins** [comment]
Great suggestion

**Reply by gmourier** [reply]
Hey, @derekperkins 👋

Thanks! Do you have a need for it?

**Reply by derekperkins** [reply]
Not for our current usage of meilisearch, but I can definitely see times where that would be useful. We use similar patterns in our other datastores today.

**Comment by Rohukas** [comment]
It'd be awesome to have this in Meilisearch!

Here’s my use case:
We currently store forum posts in Meilisearch.
Admins could access fields like user.email, while regular users wouldn’t see any of that.

Another scenario:
Our app has a feature to anonymize usernames by creating a pseudonym for each user.
We can't use Meilisearch for post-searching when anonymization is on since it could reveal the user's real name. However, we still want admins to see the names when searching with Meilisearch.

**Reply by ryanldy** [reply]
cc @gmourier

**Comment by aaroncueckermann** [comment]
Bumping this as we have a strong need for it too

**Reply by macraig** [reply]
Hello! Answered in https://github.com/orgs/meilisearch/discussions/811#discussioncomment-12466593

**Comment by sjcallender** [comment]
@gmourier - Stumbled upon this as I'm building out your exact case for the original `partner_price` example and am trying to figure out the best path. I think the best path is what you presented. :)

We're building B2B ecommerce sites in Craft Commerce and use Meilisearch search for products. As you've noted here and in https://github.com/orgs/meilisearch/discussions/811, in some cases that pricing is private and needs to be treated as a security issue.

Consider me interested. :)
